### PR TITLE
RO-1087 - [BE] remove coupon from Stripe after is_deleted is set to True

### PR DESCRIPTION
**Pull request by poxip**

### CODE REVIEW

File: aa_stripe/models.py [hunk 1]
```diff
@@ -133,6 +133,16 @@ def __str__(self):
 
     def save(self, *args, **kwargs):
         stripe.api_key = settings.STRIPE_API_KEY
+        if self.is_deleted:
```
Comment (jacoor) [inline aa_stripe/models.py]: Please check previous value of is_deleted and update it only when it was changed to true. 
Check how https://github.com/ArabellaTech/getreferd-api/blob/develop/getreferd/accounts/models.py#L40 this works here.

File: aa_stripe/models.py [hunk 2]
```diff
@@ -185,7 +200,7 @@ class StripeCharge(StripeBasicModel):
 
     def charge(self):
         if self.is_charged:
-            raise StripeMethodNotAllowed("Already charded.")
+            raise StripeMethodNotAllowed("Already charged.")
```
Comment (jacoor) [inline aa_stripe/models.py]: thanks!

File: tests/test_coupons.py [hunk 3]
```diff
@@ -149,11 +193,9 @@ def test_admin_form(self):
...
         }
         with requests_mock.Mocker() as m:
-            m.register_uri("POST", "https://api.stripe.com/v1/coupons", text=json.dumps(stripe_response))
-            m.register_uri("GET", "https://api.stripe.com/v1/coupons/25OFF", text=json.dumps(stripe_response))
-            m.register_uri("POST", "https://api.stripe.com/v1/coupons/25OFF", text=json.dumps(stripe_response))
-            coupon = StripeCoupon.objects.create(
-                coupon_id=data["coupon_id"], duration=StripeCoupon.DURATION_FOREVER, percent_off=25)
+            for method in ["GET", "POST", "DELETE"]:
+                m.register_uri(method, "https://api.stripe.com/v1/coupons/25OFF", text=json.dumps(stripe_response))
+            coupon = self._create_coupon(data["coupon_id"], amount_off=1)
```
Comment (jacoor) [inline tests/test_coupons.py]: I need more cofee. Thank you.

File: aa_stripe/models.py (hunk 1 above)
Comment (poxip) [inline aa_stripe/models.py]: Sure, will do

**Comment by jacoor** [comment]
retest this please

**Comment by jacoor** [comment]
test this please

**Pull request merged**